python_modules/dagster/dagster/_core/definitions/source_asset.py

```diff
@@ -167,7 +167,11 @@ def fn(context: OpExecutionContext) -> Output[None]:
 @experimental_param(param="io_manager_def")
 @experimental_param(param="freshness_policy")
 @experimental_param(param="tags")
-@deprecated(breaking_version="2.0.0", additional_warn_text="Use AssetSpec instead.")
+@deprecated(
+    breaking_version="2.0.0",
+    additional_warn_text="Use AssetSpec instead. If using the SourceAsset io_manager_key property, "
+    "use AssetSpec(...).with_io_manager_key(...).",
+)
```

Comment:
I think we should also augment the deprecation messaging to include information about usage of `SourceAsset` when it is an observable, but we can do that in a follow up.